packages/cozy-stack-client/src/SharingCollection.js

```diff
@@ -0,0 +1,99 @@
+import DocumentCollection, { normalizeDoc } from './DocumentCollection'
+import { uri } from './utils'
+
+const normalizeSharing = sharing => normalizeDoc(sharing, 'io.cozy.sharings')
+
+export default class SharingCollection extends DocumentCollection {
+  async findByDoctype(doctype) {
+    const resp = await this.client.fetch(
+      'GET',
+      uri`/sharings/doctype/${doctype}`
+    )
+    return {
+      ...resp,
+      data: resp.data.map(s => normalizeSharing(s))
+    }
+  }
+
+  async share(document, recipients, sharingType, description) {
+    const resp = await this.client.fetch('POST', '/sharings/', {
+      data: {
+        type: 'io.cozy.sharings',
+        attributes: {
+          description,
+          rules: getRulesFor(document, sharingType)
+        },
+        relationships: {
+          recipients: {
+            data: recipients.map(({ id, _type }) => ({ id, type: _type }))
+          }
+        }
+      }
+    })
+    return { data: normalizeSharing(resp.data) }
+  }
+
+  revokeRecipient(sharing, recipientEmail) {
+    const memberIndex = sharing.attributes.members.findIndex(
+      m => m.email === recipientEmail
+    )
+    return this.client.fetch(
+      'DELETE',
+      uri`/sharings/${sharing._id}/recipients/${memberIndex}`
+    )
+  }
+
+  revokeSelf(sharing) {
+    return this.client.fetch(
+      'DELETE',
+      uri`/sharings/${sharing._id}/recipients/self`
+    )
+  }
+}
+
+const isFile = ({ _type, type }) =>
+  _type === 'io.cozy.files' || type === 'directory' || type === 'file'
+
+const isDirectory = ({ type }) => type === 'directory'
+
+const getBehaviorsFor = (document, sharingType) => {
+  if (isFile(document) && isDirectory(document)) {
```

Comment:
Why do you check that it is both a file and a directory, even though the check is whether `type ==='directory'`?